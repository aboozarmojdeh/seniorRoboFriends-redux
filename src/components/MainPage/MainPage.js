import React, { Component, Fragment } from "react";

import "./MainPage.css";
import CardList from "../Card/CardList";
import SearchBox from "../SearchBox/SearchBox";
import Scroll from "../Scroll/Scroll";
import ErrorBoundry from '../ErrorBoundry/ErrorBoundry';
import Header from '../Header/Header';




class MainPage extends Component {
  
  //////////////async AXIOS Call ///////////////////////////
  // getUser = async (event) => {
  //   try {
  //     const response = await axios.get(
  //       "https://jsonplaceholder.typicode.com/users"
  //     );

  //     const users = response.data;
  //     this.setState({ robots: users });
  //   } catch (error) {
  //     console.error(error);
  //   }
  // };

  componentDidMount() {


    this.props.onRequestRobots();
  //   fetch("https://jsonplaceholder.typicode.com/users")
  //     .then((response) => response.json())
  //     .then((users) => this.setState({ robots: users }));
  }

  // onSearchChange = (event) => {
  //   this.setState({ searchField: event.target.value });
  // };

  filterRobots = () => {
    const { robots, searchField } = this.props;
    return robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    })
  }

  render() {
    const { robots, searchField, onSearchChange, isPending } = this.props;

    

    if (isPending) {
      return (
        <Fragment>
        <div className="tc">
        <Header />
        <SearchBox onSearchChange={onSearchChange} />
        <Scroll>
              <ErrorBoundry>
              <h1 className="f1 app-header-text">Loading!</h1>
              </ErrorBoundry>
            </Scroll>
      
      </div>
      </Fragment>
      
      );
    } else {
      return (
        <Fragment>
          <div className="tc">
          <Header />
            <SearchBox onSearchChange={onSearchChange} />
            <Scroll>
              <ErrorBoundry>
              <CardList robots={this.filterRobots()} />
              </ErrorBoundry>
            </Scroll>
          </div>
        </Fragment>
      );
    }
  }
}

export default MainPage;
