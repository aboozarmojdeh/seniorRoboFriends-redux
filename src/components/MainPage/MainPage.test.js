import React from "react";
import { shallow, mount, render } from "enzyme";

import MainPage from "./MainPage";

// npm test -- --coverage --watchAll=false

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

test("filter robots correctly", () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: "john",
        email: "john@gmail.com",
      },
    ],
    searchField: "john",
    isPending: false,
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);

  expect(wrapper2).toMatchSnapshot();
  expect(wrapper2.instance().filterRobots()).toEqual([
    {
      id: 3,
      name: "john",
      email: "john@gmail.com",
    },
  ]);
});

test("filter robots correctly2", () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: "john",
        email: "john@gmail.com",
      },
    ],
    searchField: "a",
    isPending: false,
  };
  const wrapper3 = shallow(<MainPage {...mockProps3} />);

  const filteredRobots = [];
  expect(wrapper3).toMatchSnapshot();
  expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
});

test("isPending test", () => {
  const mockProps4 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: "john",
        email: "john@gmail.com",
      },
    ],
    searchField: "a",
    isPending: true,
  };
  const wrapper4 = shallow(<MainPage {...mockProps4} />);

  const filteredRobots = [];
  expect(wrapper4).toMatchSnapshot();
  expect(wrapper4.instance().filterRobots()).toEqual(filteredRobots);
});
