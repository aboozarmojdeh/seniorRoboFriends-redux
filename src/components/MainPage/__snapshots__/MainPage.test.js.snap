// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter robots correctly 1`] = `
<Fragment>
  <div
    className="tc"
  >
    <Header />
    <SearchBox />
    <Scroll>
      <ErrorBoundry>
        <CardList
          robots={
            Array [
              Object {
                "email": "john@gmail.com",
                "id": 3,
                "name": "john",
              },
            ]
          }
        />
      </ErrorBoundry>
    </Scroll>
  </div>
</Fragment>
`;

exports[`filter robots correctly2 1`] = `
<Fragment>
  <div
    className="tc"
  >
    <Header />
    <SearchBox />
    <Scroll>
      <ErrorBoundry>
        <CardList
          robots={Array []}
        />
      </ErrorBoundry>
    </Scroll>
  </div>
</Fragment>
`;

exports[`isPending test 1`] = `
<Fragment>
  <div
    className="tc"
  >
    <Header />
    <SearchBox />
    <Scroll>
      <ErrorBoundry>
        <h1
          className="f1 app-header-text"
        >
          Loading!
        </h1>
      </ErrorBoundry>
    </Scroll>
  </div>
</Fragment>
`;
